File: backend-spring-boot/src/main/java/no/vicx/backend/config/RestClientConfig.java
package no.vicx.backend.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestClient;
import org.springframework.web.reactive.function.client.WebClient;

@Configuration
public class RestClientConfig {

    @Bean
    public RestClient restClient(RestClient.Builder builder) {
        return builder.build();
    }

    @Bean
    public WebClient webClient(WebClient.Builder builder) {
        return builder.build();
    }
}

File: backend-spring-boot/src/main/kotlin/no/vicx/backend/config/RestClientConfig.kt
package no.vicx.backend.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestClient

@Configuration(proxyBeanMethods = false)
class RestClientConfig {

    @Bean
    fun restClient(builder: RestClient.Builder): RestClient = builder.build()
}